testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -3043,10 +3051,16 @@ public void testAddColumnConcurrently()
         }
         finally {
             executor.shutdownNow();
-            executor.awaitTermination(30, SECONDS);
+            executor.awaitTermination(timeout, timeUnit);
         }
     }
 
+    @DataProvider
+    public Object[][] testAddColumnConcurrentlyConfig()
+    {
+        return new Object[][] {{30, SECONDS}};
```

Comment:
If the goal isn't to test different timeouts, this should be a constant in the class or a constructor argument, not a testing parameter.

Reply:
The issue we run into is that some databases are faster than others so the timeouts vary from subclass to subclass. One way is to have a method to provide the timeout - that works.

The reason for a DataProvider is that there are other things that we'd like to be configurable (e.g. the list of allowed exceptions for example) and having one overridable method per "configurable thing" seems wasteful and scattered.

Think of this as similar to what `testNativeLargeIn` does with DataProvider.

However I agree that in current PR since we only customize the timeout maybe having a method override makes more sense and we can see DataProvider once more customization is needed.